cmd/s3-gw/app.go

```diff
@@ -160,11 +167,32 @@ func (a *App) initLayer(ctx context.Context) {
 	}
 }
 
+func newAppSettings(log *Logger, v *viper.Viper) *appSettings {
+	policies, err := newPlacementPolicy(getDefaultPolicyValue(v), v.GetString(cfgPolicyRegionMapFile))
+	if err != nil {
+		log.logger.Fatal("failed to create new policy mapping", zap.Error(err))
+	}
+
+	return &appSettings{
+		logLevel: log.lvl,
+		policies: policies,
+	}
+}
+
+func getDefaultPolicyValue(v *viper.Viper) string {
+	defaultPolicyStr := handler.DefaultPolicy
+	if v.IsSet(cfgPolicyDefault) {
+		defaultPolicyStr = v.GetString(cfgPolicyDefault)
+	}
+
+	return defaultPolicyStr
+}
+
 func (a *App) initHandlers(ctx context.Context) {
 	a.initLayer(ctx)
 
 	var err error
-	handlerOptions := getHandlerOptions(a.cfg, a.log)
+	handlerOptions := a.getHandlerOptions()
```

Comment:
What do you think on
1. rename `a.initHandlers` to `a.initAPI` or something like that
2. define a function `a.initHandlers` where we will get handlers options and assign `handler.New` to `a.api`. 

It will make handler initialization similar to `initLayer` that seems more consistent for me.